src/main/java/org/folio/services/lines/PoLinesBatchService.java

```diff
@@ -0,0 +1,59 @@
+package org.folio.services.lines;
+
+import static org.folio.models.TableNames.PO_LINE_TABLE;
+
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.event.service.AuditOutboxService;
+import org.folio.okapi.common.GenericCompositeFuture;
+import org.folio.rest.core.models.RequestContext;
+import org.folio.rest.jaxrs.model.OrderLineAuditEvent;
+import org.folio.rest.jaxrs.model.PoLine;
+import org.folio.rest.persist.Conn;
+import org.folio.rest.persist.PostgresClient;
+
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+
+public class PoLinesBatchService {
+  private static final Logger log = LogManager.getLogger();
+  private final AuditOutboxService auditOutboxService;
+  private final PoLinesService poLinesService;
+
+
+  public PoLinesBatchService(AuditOutboxService auditOutboxService, PoLinesService poLinesService) {
+    this.auditOutboxService = auditOutboxService;
+    this.poLinesService = poLinesService;
+  }
+
+  public Future<Void> poLinesBatchUpdate(List<PoLine> poLines, PostgresClient pgClient, Map<String, String> okapiHeaders,
+    Context vertxContext) {
+
+    if (CollectionUtils.isEmpty(poLines)) {
+      log.warn("poLinesBatchUpdate:: po line list is empty");
+      return Future.succeededFuture();
+    }
+
+    return pgClient.withTrans(conn ->
+      conn.updateBatch(PO_LINE_TABLE, poLines)
+        .compose(rowSet -> updatePoLinesWithTitle(conn, poLines, okapiHeaders, vertxContext))
+        .compose(rowSet -> auditOutboxService.saveOrderLinesOutboxLogs(conn, poLines, OrderLineAuditEvent.Action.EDIT, okapiHeaders))
+        .mapEmpty()
+    );
+
+  }
+
+  private CompositeFuture updatePoLinesWithTitle(Conn conn, List<PoLine> poLines, Map<String, String> okapiHeaders, Context vertxContext) {
+    var futures = poLines.stream()
+      .filter(poLine -> !poLine.getIsPackage())
+      .map(poLine -> poLinesService.updatePoLineWithTitle(conn, poLine.getId(), poLine, new RequestContext(vertxContext, okapiHeaders)))
```

Comment:
This method is updating poline at second time(first time poline we updated previously using conn.updateBatch).
If con.upsertBatch is impossible to use for creating/updating batch titles we should use poLinesService.updateTitle() instead of poLinesService.updatePoLineWithTitle()